database/database.go

```diff
@@ -98,13 +99,41 @@ func (d *Database) GetUser(username string) (user model.User, err error) {
 	return user, nil
 }
 
+// GetUserInfo retrieves metadata about a user.
+func (d *Database) GetUserInfo(userID string) (userInfo model.UserInfo, err error) {
+	userInfo = model.UserInfo{}
+
+	sqlStatement := `
+		SELECT COUNT(t), COUNT(tp), MAX(tp.postedat)
+			FROM board.thread t
+				INNER JOIN board.thread_post tp ON t.id = tp.threadid
```

Comment:
This could get a bit costly, let's keep an eye on it. It potentially joins a BIG table. We might be able to do some sort of trigger or etc... to update this info.

Reply:
Okay, that sounds good.